x-pack/plugins/data_visualizer/public/application/index_data_visualizer/ui_actions/create_field_stats_table.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { PresentationContainer } from '@kbn/presentation-containers';
+import type { EmbeddableApiContext } from '@kbn/presentation-publishing';
+import type { UiActionsActionDefinition } from '@kbn/ui-actions-plugin/public';
+import { IncompatibleActionError } from '@kbn/ui-actions-plugin/public';
+import type { CoreStart } from '@kbn/core-lifecycle-browser';
+import { FIELD_STATS_EMBEDDABLE_TYPE } from '../embeddables/field_stats/constants';
+import type { DataVisualizerStartDependencies } from '../../common/types/data_visualizer_plugin';
+import type { FieldStatisticsTableEmbeddableApi } from '../embeddables/field_stats/types';
+
+const parentApiIsCompatible = async (
+  parentApi: unknown
+): Promise<PresentationContainer | undefined> => {
+  const { apiIsPresentationContainer } = await import('@kbn/presentation-containers');
+  // we cannot have an async type check, so return the casted parentApi rather than a boolean
+  return apiIsPresentationContainer(parentApi) ? (parentApi as PresentationContainer) : undefined;
+};
+
+interface FieldStatsActionContext extends EmbeddableApiContext {
+  embeddable: FieldStatisticsTableEmbeddableApi;
+}
+export function createAddFieldStatsTableAction(
+  coreStart: CoreStart,
+  pluginStart: DataVisualizerStartDependencies
+): UiActionsActionDefinition<FieldStatsActionContext> {
+  return {
+    id: 'create-field-stats-table',
+    getIconType: () => 'inspect',
```

Comment:
@joana-cps any thoughts on this icon?

<img width="322" alt="Screenshot 2024-06-10 at 13 42 53" src="https://github.com/elastic/kibana/assets/7405507/8363485b-1485-45bd-aa57-4147cabd4f2f">

Reply:
This icon is already in use in dashboards to inspect elements, I'll draft up a new one more tailored for field statistics. Added to this [UX issue](https://github.com/elastic/platform-ux-team/issues/302) in progress